industrial_extrinsic_cal/src/check_if_points_in_pic.cpp

```diff
@@ -0,0 +1,349 @@
+  string targetFile = "/home/lawrencelewis/catkin_ws/src/industrial_calibration/industrial_extrinsic_cal/yaml/ical_srv_target.yaml";
+  vector<boost::shared_ptr<Target>>  myYamlDefinedTargets;
+  if(!parseTargets(targetFile , myYamlDefinedTargets))
+{
+    ROS_ERROR("file name bad");
+    return 0;
+  }
+   boost::shared_ptr<Target> Mytarget = myYamlDefinedTargets[0];
+
+  //finds the max values to be used as corner points
+  ROS_INFO("target name = %s num_points = %d size of pts_ = %d",Mytarget->target_name_,Mytarget->num_points_,(int)Mytarget->pts_.size());
+  xMax = Mytarget->pts_[0].x;
+  xMin = Mytarget->pts_[0].x;
+  yMax = Mytarget->pts_[0].y;
+  yMin = Mytarget->pts_[0].y;
+  for(int i=0;i<Mytarget->num_points_;i++)
+  {
+    if(Mytarget->pts_[i].x > xMax) xMax = Mytarget->pts_[i].x;
+    if(Mytarget->pts_[i].y > yMax) yMax = Mytarget->pts_[i].y;
+    if(Mytarget->pts_[i].x < xMin) xMin = Mytarget->pts_[i].x;
+    if(Mytarget->pts_[i].y < yMin) yMin = Mytarget->pts_[i].y;
+  }
+  Eigen::Vector3d corner_points[4];
+
+  corner_points[0] = Eigen::Vector3d((xMin-xMax/2)/5,(yMin-yMax/2)/5,0);
+  corner_points[1] = Eigen::Vector3d((xMax-xMax/2)/5,(yMin-yMax/2)/5,0);
+  corner_points[2] = Eigen::Vector3d((xMin-xMax/2)/5,(yMax-yMax/2)/5,0);
+  corner_points[3] = Eigen::Vector3d((xMax-xMax/2)/5,(yMax-yMax/2)/5,0);
+
+  center_point_of_target = finds_middle_of_target(corner_points, center_Of_TargetX,center_Of_TargetY);
+
+  // creates camera transforms in the shape of a cone
+  ros::Publisher pub = n.advertise<geometry_msgs::PoseArray>("topic", 1, true);
+  // radius of the cone = (poseHeight/ std::tan(angleOfCone))
+  geometry_msgs::PoseArray msg = create_all_poses(poseHeight, spacing_in_z, angleOfCone, numberOfStopsForPhotos,  center_point_of_target );
+
+  //creates rectanglular target in rviz
+  initialization.create_rviz_target(corner_points,msg);
+
+  // convert the messages back into poses
+  EigenSTL::vector_Affine3d AllcameraPoses(msg.poses.size());
+  geometry_msgs::PoseArray msg2;
+  for(int i=0; i< msg.poses.size(); i++)
+  {
+    tf::poseMsgToEigen(msg.poses[i],AllcameraPoses[i]);
+  }
+
+  // filters out unreachable and unseeable poses
+  geometry_msgs::PoseArray filtered_msgs = pose_filters(msg2,tf_transform, image_width, image_height, AllcameraPoses, corner_points, fx,  fy,  cx,  cy );
+
+
+  ros::Publisher pub2 = n.advertise<geometry_msgs::PoseArray>("topic2", 1, true);
+  filtered_msgs.header.frame_id = "target";
+  filtered_msgs.header.stamp = ros::Time::now();
+
+  // publishes all transforms
+  ros::Rate r(10); // 10 hz
+  while (ros::ok())
+  {
+    pub2.publish(filtered_msgs);
+    pub.publish(msg);
+    ros::spinOnce();
+    r.sleep();
+  }
+} //end of main
+
+// Transforms a point into the camera frame then projects the new point into the 2d screen
+void imagePoint(Eigen::Vector3d TargetPoint , double fx, double fy, double cx, double cy, double &u, double &v,Eigen::Affine3d cameraPose )
```

Comment:
these helper functions could either be in their own cpp file, or private members of class foo.